java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitution.java

```diff
@@ -109,6 +109,13 @@ public TypeSubstitution combine(TypeSubstitution source) {
         result.add(typeVar, targetType);
         continue;
       }
+      if(targetType == substitutedType) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)